src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py

```diff
@@ -293,8 +299,39 @@ def connection_create(cmd, client,  # pylint: disable=too-many-locals,too-many-s
                       namespace=None,                                        # Resource.EventHub
                       webpubsub=None,                                        # Resource.WebPubSub
                       signalr=None):                                         # Resource.SignalR
-    from ._credential_free import enable_mi_for_db_linker
-
+    all_auth_info = []
+    if secret_auth_info is not None:
+        all_auth_info.append(secret_auth_info)
+    if secret_auth_info_auto is not None:
+        all_auth_info.append(secret_auth_info_auto)
+    if user_identity_auth_info is not None:
+        all_auth_info.append(user_identity_auth_info)
+    if system_identity_auth_info is not None:
+        all_auth_info.append(system_identity_auth_info)
+    if service_principal_auth_info_secret is not None:
+        all_auth_info.append(service_principal_auth_info_secret)
+    if not new_addon and len(all_auth_info) != 1:
+        raise ValidationError('Only one auth info is needed')
```

Comment:
duplicate logics also in `local_connection_create`, `connection_create_func`, and `local_connection_create_func`consider doing some refractoring to reuse

Reply:
fixed